ts/src/profiler.ts

```diff
@@ -72,6 +75,29 @@ export interface RequestProfile {
   labels?: {instance?: string};
 }
 
+/**
+ * @return number indicated by backoff if the response indicates a backoff and
+ * that backoff is greater than 0. Otherwise returns undefined.
+ */
+// tslint:disable-next-line: no-any
+function getServerResponseBackoff(response: any): number|undefined {
+  if (response && response.body && response.body.error &&
+      response.body.error.details &&
+      response.body.error.details instanceof Array) {
```

Comment:
I'm not sure that logging a warning whenever a backoff is not specified in this format makes sense, since a server-specified backoff is only expected when 

I do log the backoff [here](https://github.com/nolanmar511/cloud-profiler-nodejs/blob/1fc8cbf858618db282a35239d649125e37e30154/ts/src/profiler.ts#L313), and log errors differently depending on whether or not the error has a backoff. Would this be sufficient for observing if there is a problem with getting the backoff?

Reply:
No that log doesn't really address my concerns. I am not worried about the backoff but rather the weak typing in JavaScript. There is no easy way for me, as a code reviewer, to verify that the code here matches what actual real server response looks like.

I would be less concerned if this was build on top of stronger types - say provided by #87. I would be okay if you want to address this as a follow-on.